src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -223,6 +223,61 @@
       <summary>Creates a PointerTypeSyntax node.</summary>
     </FactoryComment>
   </Node>
+  <Node Name="FunctionPointerTypeSyntax" Base="TypeSyntax">
+    <Kind Name="FunctionPointerType"/>
+    <Field Name="DelegateKeyword" Type="SyntaxToken">
+      <Kind Name="DelegateKeyword"/>
+      <PropertyComment>
+        <summary>SyntaxToken representing the delegate keyword.</summary>
+      </PropertyComment>
+    </Field>
+    <Field Name="AsteriskToken" Type="SyntaxToken">
+      <Kind Name="AsteriskToken"/>
+      <PropertyComment>
+        <summary>SyntaxToken representing the asterisk.</summary>
+      </PropertyComment>
+    </Field>
+    <Field Name="CallingConvention" Type="SyntaxToken" Optional="true">
+      <Kind Name="CdeclKeyword" />
+      <Kind Name="ManagedKeyword" />
+      <Kind Name="StdcallKeyword" />
+      <Kind Name="ThiscallKeyword" />
+      <Kind Name="UnmanagedKeyword" />
```

Comment:
>Kind [](start = 7, length = 4)

I don't think this list belongs in the parser.  I think it belongs in semantic analysis.  The parser should accept any identifier and let semantic analysis sort it out.  I could be wrong about this...

Reply:
IDE would def like nice kinds for contextual keywords fwiw.  If these really are well-known names that the compiler cares about, then this does make sense to me.  I don't feel super strongly about it though.

Reply:
Some people don't like "overfitting of the semantic model" ;)

---
In reply to: [348125579](https://github.com/dotnet/roslyn/pull/39879#discussion_r348125579) [](ancestors = 348125579)

Reply:
> The parser should accept any identifier and let semantic analysis sort it out.

This is what I've implemented, as I consume any other identifier and replace with a missing token.

---
In reply to: [348138189](https://github.com/dotnet/roslyn/pull/39879#discussion_r348138189) [](ancestors = 348138189,348125579)

Reply:
> Some people don't like "overfitting of the semantic model" ;)

Good point.  I would be ok with this being just an identifier and being purely semantic checked by the compiler.